src/Scriban/Parsing/Lexer.cs

```diff
@@ -822,6 +822,15 @@ private bool ReadCode()
                         NextChar();
                         break;
                     }
+                    else
+                    {
+                       if (IsRemainderOfLineWhitespace())
```

Comment:
I don't think it should be handled at the Lexer level.

It should be instead handled when parsing the token at this line in Parser.Expressions.cs

```
pipeCall.PipeToken = ParseToken(Current.Type); // skip | or |>
```

You should maybe add a boolean to ParseToken to allow to eat newlines and in ParseToken, use `_allowNewLineLevel++` and restore it right after the call to get a next token Next().

Reply:
oh - looks like our comments crossed over

Reply:
`_allowNewLineLevel` is used today to allow nested new lines in object initializers and array initializers, but we can use it for other purposes.

Reply:
Oh - didn't realise it would be so easy....  The current change also allows the Scientific pipe operator to now span lines I think.  Is that desirable or is it going to break some assumption in Scientific mode?

Reply:
Yeah, that should be fine for scientific mode as well